Sources/Get/DataLoader.swift

```diff
@@ -337,30 +337,31 @@ extension Optional: OptionalDecoding {}
 func encode(_ value: Encodable, using encoder: JSONEncoder) async throws -> Data? {
     if let data = value as? Data {
         return data
-    } else if let string = value as? String {
+    }
+    if let string = value as? String {
         return string.data(using: .utf8)
-    } else {
-        return try await Task.detached {
-            try encoder.encode(AnyEncodable(value: value))
-        }.value
     }
+    return try await Task.detached {
+        try encoder.encode(AnyEncodable(value: value))
+    }.value
 }
 
 func decode<T: Decodable>(_ data: Data, using decoder: JSONDecoder) async throws -> T {
-    if data.isEmpty, T.self is OptionalDecoding.Type {
-        return Optional<Decodable>.none as! T
-    } else if T.self == Data.self {
-        return data as! T
-    } else if T.self == String.self {
-        guard let string = String(data: data, encoding: .utf8) else {
+    if data.isEmpty, T.self is OptionalDecoding.Type, let value = Optional<Decodable>.none as? T {
+        return value
+    }
+    if T.self == Data.self, let value = data as? T {
```

Comment:
I don't think checking for the type first is needed anymore, `let value = data as? T` should do the work already.